### PR TITLE
deploy artifact to webapp & ui flaws fix.

### PR DESCRIPTION
**Pull request by wangmingliang-ms**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/subscription/SubscriptionComboBox.java [hunk 1]
```diff
@@ -31,6 +31,13 @@
 import java.util.Objects;
 
 public class SubscriptionComboBox extends AzureComboBox<Subscription> {
+
+    public SubscriptionComboBox() {
+        super();
+        final List<Subscription> items = AzureMvpModel.getInstance().getSelectedSubscriptions();
+        this.setValue(items.get(0)); // select the first subscription
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/subscription/SubscriptionComboBox.java:38]: avoid IndexOutOfBoundsException

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureComboBox.java [hunk 2]
```diff
@@ -102,6 +104,30 @@ public void customize(@NotNull final JList<? extends T> l, final T t, final int
...
+        return ((T) this.getSelectedItem());
+    }
+
+    @Override
+    public void setValue(final T val) {
+        this.value = val;
+        final List<T> items = this.getItems();
+        if (items.contains(this.value)) {
+            super.setSelectedItem(this.value);
+        } else if (items.size() > 0) {
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureComboBox.java:118]: !isEmpty

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/subscription/SubscriptionComboBox.java (hunk 1 above)
Comment (wangmingliang-ms) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/subscription/SubscriptionComboBox.java:38]: user can not come here if he doesn't have any subscriptions.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/webapp/WebAppConfigFormPanelBasic.java [hunk 3]
```diff
@@ -112,7 +113,9 @@ public WebAppConfig getData() {
         plan.setRegion(region);
         config.setServicePlan(plan);
 
-        config.setApplication(path);
+        if (Objects.nonNull(artifact)) {
+            config.setApplication(Paths.get(artifact.getTargetPath()));
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/webapp/WebAppConfigFormPanelBasic.java:117]: AzureArtifactManager.getInstance(project).getFileForDeployment

**Review by andxu — APPROVED** [review]

**Pull request merged**